### PR TITLE
Worked on the issue #2084 --> Fix Fossology agent tests. tests_common…

### PR DESCRIPTION
**Pull request by mridul45**
…_menu.php

## Description

Please describe the changes in your pull request in few words here.

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

### CODE REVIEW

File: src/lib/php/tests/test_common_menu.php [hunk 1]
```diff
@@ -1,138 +1,112 @@
 <?php
-/*
- SPDX-FileCopyrightText: © 2011 Hewlett-Packard Development Company, L.P.
 
- SPDX-License-Identifier: GPL-2.0-only
-*/
```
Comment (GMishx) [inline src/lib/php/tests/test_common_menu.php:6]: Please do not remove copyrights.

File: src/lib/php/tests/test_common_menu.php [hunk 2]
```diff
@@ -1,138 +1,112 @@
...
+namespace YourNamespace\Tests;
```
Comment (GMishx) [inline src/lib/php/tests/test_common_menu.php:3]: Please use better namespace name.

File: src/lib/php/tests/test_common_menu.php [hunk 3]
```diff
@@ -1,138 +1,112 @@
...
 
-/**
- * \file test_common_menu.php
- * \brief unit tests for common-menu.php
- */
-
```
Comment (GMishx) [inline src/lib/php/tests/test_common_menu.php:12]: Please do not remove doc string documenting the file.

File: src/lib/php/tests/test_common_menu.php [hunk 4]
```diff
@@ -1,138 +1,112 @@
...
-use PHPUnit\Runner\Version as PHPUnitVersion;
+use PHPUnit\Framework\TestCase;
 
 require_once(dirname(__FILE__) . '/../common-menu.php');
 require_once(dirname(__FILE__) . '/../common-parm.php');
 
 /**
- * \class test_common_menu
+ * Test suite for common-menu.php
```
Comment (GMishx) [inline src/lib/php/tests/test_common_menu.php:11]: Please use `\class` to document the class.

File: src/lib/php/tests/test_common_menu.php [hunk 5]
```diff
@@ -1,138 +1,112 @@
...
+    public function testMenuEndlessPage()
+    {
+        $page = 10;
+        $uri = "http://fossology.org/repo/";
+        $expected = "<a class='page-link' href='http:\/\/fossology.org\/repo\/&page=9'>Prev<\/a>";
+        $result = MenuEndlessPage($page, 1, $uri);
+
+        // Assert that the result contains the expected output
+        $this->assertStringContainsString("<a class='page-link' href='#'>11<\/a>", $result);
+        $this->assertStringContainsString($expected, $result);
```
Comment (GMishx) [inline src/lib/php/tests/test_common_menu.php:43]: Please use 2 space indentation. Read [coding guidelines](https://github.com/fossology/fossology/wiki/Coding-Style)

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
@mridul45 , please fix the commit message. Please follow the commit convention mentioned in [contributing guide](https://github.com/fossology/fossology/blob/2ce7d20684a1e5b3dd14d042cd92b3afa034f6c8/CONTRIBUTING.md#git-commit-conventions)

**Comment by mridul45** [comment]
ok sir

**Comment by GMishx** [comment]
Closing in favor of #2698

**Pull request closed without merging**